modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts

```diff
@@ -139,9 +139,9 @@ export class ParquetEnvelopeReader {
 
     let dictionary;
 
-    const dictionaryPageOffset = colChunk?.meta_data?.dictionary_page_offset;
+    const dictionaryPageOffset = Number(colChunk?.meta_data?.dictionary_page_offset);
```

Comment:
The test case is called dictionary--offset-zero but we don't read a dictionary when offset zero? 

Can we add more comments to code?

Also it is a little bit suspect to rely on number conversion to convert undefined to 0. Maybe we could be more explicit?

Reply:
1. Dictionary with offset 0 file created by` parquet-mr version 1.12.0-201812210311360288-a86293f (build cec1a483e9dcd545e09170ae787d3dcb13744433)` tool. I take it from metadata while debugging process. And it has differences with Impala parquet format in `dictionary_page_offset` and `data_page_offset` fields. It described [here](https://stackoverflow.com/questions/55225108/why-is-dictionary-page-offset-0-for-plain-dictionary-encoding).
Actually it works as it is in our code and I don't fully understand how. The interesting thing that we don't get a dictionary at all but all pages are read as `DATA_PAGE` and has correct values if we compare with` parquet-tools` results for this file.
2. I can describe that can be pages with dictionary offset 0 and it is `parquet-mr` format.
3. Actually yes it is wrong. If `colChunk?.meta_data?.dictionary_page_offset` is `undefined` that means no `dictionaryPage` at all, but if it is 0 looks like it is `parquet- mr` file and we should handle it properly.

Reply:
@ibgreen What I need now is to find out how to get `dictionary` from column chunk in `parquet-mr` files.